packages/components/src/text/utils.ts

```diff
@@ -41,9 +59,10 @@ import { createElement } from '@wordpress/element';
  * @return {{[K in keyof T as Lowercase<string & K>]: T[K]}} The mapped props.
  */
 /* eslint-enable jsdoc/valid-types */
-const lowercaseProps = ( object ) => {
-	/** @type {any} */
-	const mapped = {};
+const lowercaseProps = < T extends Record< string, unknown > >( object: T ) => {
+	// TODO: Replace with more specific types
```

Comment:
A quick way to make types work as before, without TypeScript errors, and all while removing previous JSDocs could be to do something like this:

```ts
/**
 * Maps props to lowercase names.
 *
 * @param object Props to map.
 * @return The mapped props.
 */
const lowercaseProps = < T extends Record< string, unknown > >( object: T ) => {
	const mapped: Record< string, unknown > = {};
	for ( const key in object ) {
		mapped[ key.toLowerCase() ] = object[ key ];
	}
	return mapped as { [ K in keyof T as Lowercase< string & K > ]: T[ K ] };
};
```

Reply:
Done.